lib/rubocop/cop/style/numbered_parameters_limit.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop detects use of an excessive amount of numbered parameters in a
+      # single block. Having too many numbered parameters can make code too
+      # cryptic and hard to read.
+      #
+      # @example
+      #   # bad
+      #   foo { _1.call(_2, _3, _4) }
+      #
+      #   # good
+      #   foo { do_something(_1) }
+      class NumberedParametersLimit < Base
+        extend TargetRubyVersion
+        extend ExcludeLimit
+
+        minimum_target_ruby_version 2.7
+        exclude_limit 'Max'
+
+        MSG = 'Avoid using more than %<max>i numbered %<parameter>s; %<count>i detected.'
+
+        def on_numblock(node)
+          _send_node, param_count, * = *node
+          return if param_count <= max_count
+
+          parameter = max_count > 1 ? 'parameters' : 'parameter'
+          message = format(MSG, max: max_count, parameter: parameter, count: param_count)
+          add_offense(node, message: message) { self.max = param_count }
+        end
+
+        private
+
+        def max_count
+          # There cannot be more than 9 numbered parameters
+          [cop_config['Max'], 9].min
```

Comment:
It may be safer to use `cop_config.fetch('Max', DEFAULT_MAX_VALUE)`.

```diff
+          DEFAULT_MAX_VALUE = 1

...

-          # There cannot be more than 9 numbered parameters
-          [cop_config['Max'], 9].min
+          max = cop_config.fetch('Max', DEFAULT_MAX_VALUE)
+          
+          # There cannot be more than 9 numbered parameters.
+          [max, 9].min
```

Reply:
Updated, thanks!